File: js/nova/datastruct/list/ListNode.js
var ListNode = function () {
	var self = this;
	
	this.data = nova_null;
	this.next = nova_null;
	
	this.ListNode = function (data) {
		self.this(data);
	};
	
	this.destroy = function () {
	};
	
	this.this = function (data) {
		self = data;
	};
	
	this.clone = function () {
		var clone;
		clone = new ListNode(self.data);
		clone.next = self.next;
		return clone;
	};
	
	this.super = function () {
	};
	
	
};

File: java/nova/datastruct/list/ListNode.java
package nova.datastruct.list;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;

public class ListNode extends NovaObject
{
	
	public NovaObject data;
	public ListNode next;
	
	
	
	
	public ListNode(NovaObject data)
	{
		init(data);
	}
	
	public void init(NovaObject data)
	{
		this.data = data;
	}
	
	public ListNode clone()
	{
		ListNode clone;
		clone = new ListNode(data);
		clone.next = next;
		return clone;
	}
	
}